### PR TITLE
compiler: Add GuardOverflow

### PR DESCRIPTION
**Pull request by FabioLuporini**
Used in DevitoPRO but added here because 1) why not and 2) I suspect I'm going to use it in the near future for other stuff

### CODE REVIEW

**Review by mloubout — APPROVED** [review]

File: devito/ir/support/guards.py [hunk 1]
```diff
@@ -194,5 +214,6 @@ def transform_guard(expr, guard_type, callback):
     GuardBoundLe: GuardBoundGt,
     GuardBoundGt: GuardBoundLe,
     GuardBoundNextLe: GuardBoundNextGt,
-    GuardBoundNextGt: GuardBoundNextLe
+    GuardBoundNextGt: GuardBoundNextLe,
+    GuardOverflow: GuardOverflowLt
```
Comment (georgebisbas) [inline devito/ir/support/guards.py:218]: I think for homogeneity we should have:

GuardOverflowGe: GuardOverflowLt
GuardOverflowLt: GuardOverflowGe

**Review by georgebisbas — COMMENTED** [review]
Two very small-small-minor comments, otherwise GTG

File: devito/ir/support/guards.py [hunk 2]
```diff
@@ -4,15 +4,15 @@
 (e.g., Eq, Cluster, ...) should be evaluated at runtime.
 """
 
-from sympy import Le, Gt, Mul, true
+from sympy import Le, Lt, Ge, Gt, Mul, true
```
Comment (georgebisbas) [inline devito/ir/support/guards.py:7]: AB order?

File: devito/ir/support/guards.py (hunk 1 above)
Comment (FabioLuporini) [reply devito/ir/support/guards.py]: fixed

**Review by georgebisbas — APPROVED** [review]

**Pull request merged**